API.md

```diff
@@ -0,0 +1,121 @@
+API Proposal
+============
+## Use Cases
+- **Application Administration** (e.g. GPII): updating multiple settings of multiple (other) applications
+- **Application Settings** (e.g. my app): read/update/monitor applications's own settings
+
+
+## To throw or not to throw
+
+## Module Styles
+[Interesting read-up concerning nodejs module design](http://psteeleidem.com/node-module-factory-pattern/)
+
+### 1) Module style - Namespaced Collection of functions
```

Comment:
i agree that the 3rd sytle (OO with 'new') is ugly (for the CON reasons I listed) - so I guess that one is out.

But I still feel like using the module style (1st syle) doesn't do it justice. I agree that looking at the orignal shell/C api for naming is useful. BUT, replicating the C/Shell api one to one in javascript just because Shell/C does not have a understanding of OO is wrong IMO. Javascript is an OO language and looking at the C api calls like `g_settings_new()` already indicates an OO view on the domain. Also, nearly all C api calls that manipulate gsettings do something like this `g_settings_set_value (settings, key, value);`. This is a very procedural style like C, PHP (used to be).

The OO style would not take the settings object as an argument but operate on it e.g. `settings.set_value(key,value)` - IMO this is more like javascript does things e.g. `myArray.reverse()`, `myString.replace(/xyz/, 'abc')`

I would like to have something like the 2nd style (module with factory method) without including the module methods of style 1.

I think the problem is that we try to cater to two different use cases.

Combining style 1 and 2 would be a compromise. But it would need decent naming to not confuse people. Again... not perfect.

I think further discussion is needed.